x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/deprecation/DeprecationIssue.java

```diff
@@ -175,4 +185,103 @@ public int hashCode() {
     public String toString() {
         return Strings.toString(this);
     }
+
+    public static Map<String, Object> createMetaMapForRemovableSettings(List<String> removableSettings) {
+        Map<String, Object> actionsMap = new HashMap<>();
+        actionsMap.put(ACTION_TYPE, REMOVE_SETTINGS_ACTION_TYPE);
+        actionsMap.put(OBJECTS_FIELD, removableSettings);
+        return Collections.singletonMap(ACTIONS_META_FIELD, Collections.singletonList(Collections.unmodifiableMap(actionsMap)));
+    }
+
+    /**
+     * This method returns a DeprecationIssue that has in its meta object the intersection of all auto-removable settings that appear on
+     * all of the DeprecationIssues that are passed in. This method assumes that all DeprecationIssues passed in are equal, except for the
+     * auto-removable settings in the meta object.
+     * @param similarIssues
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    public static DeprecationIssue getIntersectionOfRemovableSettings(List<DeprecationIssue> similarIssues) {
```

Comment:
This function feels really large, could you split it up into smaller pieces? Maybe a separate part for retrieving the leastCommonRemovableSettings?

Reply:
Done.